### ISSUE TITLE
eks requst sts timeout.

### ISSUE THREAD

**Issue by tolerant-hu** [kubernetes-sigs/aws-load-balancer-controller#2477]
**Describe the bug**
aws-load-balancer-controller requst https://sts.ap-southeast-1.amazonaws.com timeout.

**Steps to reproduce**

1. eksctl create cluster --name <> --without-nodegroup
2. eksctl create nodegroup --cluster <> --name <> ....
3. step by step exec to https://docs.aws.amazon.com/eks/latest/userguide/aws-load-balancer-controller.html
4. end

```
kubectl get pod -n kube-system
aws-load-balancer-controller-66c8bdc46-7vgrb   1/1     Running   0          
aws-load-balancer-controller-66c8bdc46-j8r9w   1/1     Running   0          
```

**Expected outcome**

1. I try to test load-balancer-controller
```
kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/aws-load-balancer-controller/v2.3.1/docs/examples/2048/2048_full.yaml
```
aws-load-balancer-controller return logs:

```
kubectl get ing -n game-2048
NAME           CLASS    HOSTS   ADDRESS   PORTS   AGE
ingress-2048   <none>   *                 80      3m53s
```

```
{"level":"error","ts":1643192131.4378264,"logger":"controller-runtime.manager.controller.ingress","msg":"Reconciler error","name":"ingress-2048","namespace":"game-2048","error":"WebIdentityErr: failed to retrieve credentials\ncaused by: RequestError: send request failed\ncaused by: Post \"https://sts.ap-southeast-1.amazonaws.com/\": dial tcp 103.246.148.245:443: i/o timeout"}
```

2. I try to check network on cluster node
```
# on node
root@ip-xxxx:~# docker run -it --rm centos /bin/bash
[root@f6dff7c9af15 /]# yum install telnet
[root@f6dff7c9af15 /]# telnet sts.ap-southeast-1.amazonaws.com 443
Trying 103.246.148.230...
Connected to sts.ap-southeast-1.amazonaws.com.
Escape character is '^]'.
```

**Environment**

* AWS Load Balancer controller version: "GitVersion":"v2.3.1","GitCommit":"1d492cb8648b2053086761140d9db9236f867237"
* Kubernetes version: GitVersion:"v1.21.5-eks-bc4871b", GitCommit:"5236faf39f1b7a7dabea8df12726f25608131aa9"
* Using EKS (yes)

**Additional Context**:

**Comment by M00nF1sh** [comment]
@tolerant-hu 
in your tests, you are using `docker run` on a node, which is using node's network instead of network setup by CNI(which pods is using).
Would you help try setup a Kubernetes pod(such as using AL2 image) and test connectivity to sts by execute into that pod(kubectl exec)
I guess it's some misconfigured setting that cause this , such as enabledExternalSNAT on cni but nodes are in public subnet.

**Comment by kishorj** [comment]
@tolerant-hu, I'm closing the issue. If you have further concerns, feel free to reopen or create a new one.

**Comment by talkerbox** [comment]
@tolerant-hu did you fix this error? I got the same today, and even checked connectivity from the pod on the same cluster and same namespace

UPD. Actually, looks like my test also was incorrect. I ran the node-shell pod via k9s, and it has the different network and NS resolver configuration, so everything works fine. But when the real pod has been deployed (via kubectl) - I got the same error. After some investigations I noticed that there wasn't `core-dns` addon in my EKS cluster. I added it - and everything works fine now.
